AU2/plugins/custom_plugins/CompetencyPlugin.py

```diff
@@ -139,26 +170,71 @@ def on_hook_respond(self, hook: str, htmlResponse, data) -> List[HTMLComponent]:
         return []
 
     def on_event_request_create(self) -> List[HTMLComponent]:
-        return [
-            Dependency(
-                dependentOn="CorePlugin_assassin_pseudonym",
-                htmlComponents=[
-                    AssassinDependentIntegerEntry(
-                        pseudonym_list_identifier="CorePlugin_assassin_pseudonym",
-                        identifier=self.html_ids["Competency"],
-                        title="Extend competency?",
-                        default={},
-                        global_default=GENERIC_STATE_DATABASE.arb_int_state.get(self.plugin_state["DEFAULT"], DEFAULT_EXTENSION)
-                    )
-                ]
-            )
-        ]
+        if GENERIC_STATE_DATABASE.arb_state.get(self.plugin_state["Auto Competency"], "Manual") != "Full Auto":
+            return [
+                Dependency(
+                    dependentOn="CorePlugin_assassin_pseudonym",
+                    htmlComponents=[
+                        AssassinDependentIntegerEntry(
+                            pseudonym_list_identifier="CorePlugin_assassin_pseudonym",
+                            identifier=self.html_ids["Competency"],
+                            title="Extend competency?",
+                            default={},
+                            global_default=GENERIC_STATE_DATABASE.arb_int_state.get(self.plugin_state["DEFAULT"], DEFAULT_EXTENSION)
+                        )
+                    ]
+                )
+            ]
+        else:
+            return []
 
     def on_event_create(self, e: Event, htmlResponse) -> List[HTMLComponent]:
-        e.pluginState.setdefault(self.identifier, {})[self.plugin_state["COMPETENCY"]] = htmlResponse[self.html_ids["Competency"]]
-        return [Label("[COMPETENCY] Success!")]
+        message = []
+        competency_extension = GENERIC_STATE_DATABASE.arb_int_state.get(self.plugin_state["DEFAULT"], DEFAULT_EXTENSION)
+        if GENERIC_STATE_DATABASE.arb_state.get(self.plugin_state["Auto Competency"], "Manual") != "Full Auto":
+            e.pluginState.setdefault(self.identifier, {})[self.plugin_state["COMPETENCY"]] = htmlResponse[self.html_ids["Competency"]]
+        else:  # Create an empty competency dict when nothing was entered (due to full auto mode)
+            e.pluginState.setdefault(self.identifier, {})[self.plugin_state["COMPETENCY"]] = {}

```

Comment:
AU2 should already be robust to plugins being off. 

```suggestion
```

If you really need it setdefault through all control paths, move the body of the `else` clause before the `if` and update the `if` clause to be `e.pluginState[self.identifier][self. ...] = htmlResponse[...]`